ccl/streamingccl/streamingest: TestTenantStreamingRetryLoadJob failed

ccl/streamingccl/streamingest.TestTenantStreamingRetryLoadJob [failed](https://tanzanite.cluster.engflow.com/invocation/5fcd85ea-fc1a-474b-b381-6221ecf6c594) on master @ [14733120c08ba1a0a050bb5442c3311090206566](https://github.com/cockroachdb/cockroach/commits/14733120c08ba1a0a050bb5442c3311090206566):


```
=== RUN   TestTenantStreamingRetryLoadJob
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestTenantStreamingRetryLoadJob2671980607
    test_log_scope.go:81: use -show-logs to present logs inline

pkg/ccl/streamingccl/replicationtestutils/testutils.go:150: ((*TenantStreamingClusters).init)
	NOTICE: .WaitForTenantCapabilities() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().WaitForTenantCapabilities() instead.
    testutils.go:489: condition failed to evaluate within 45s: from testutils.go:501: waiting for stream ingestion job progress 1700384533.468934036,0 to advance beyond 1700384534.008791287,0
    panic.go:523: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantStreamingRetryLoadJob2671980607
--- FAIL: TestTenantStreamingRetryLoadJob (56.79s)
```
<p>Parameters: <code>attempt=1</code>
, <code>run=24</code>
, <code>shard=4</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStreamingRetryLoadJob.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-33642

--- a/pkg/ccl/streamingccl/streamingest/replication_stream_e2e_test.go
+++ b/pkg/ccl/streamingccl/streamingest/replication_stream_e2e_test.go
@@ -1143,6 +1143,11 @@ func TestTenantStreamingRetryLoadJob(t *testing.T) {
 	c, cleanup := replicationtestutils.CreateTenantStreamingClusters(ctx, t, args)
 	defer cleanup()
 
+	// Inject an error to fail the resumer.
+	mu.Lock()
+	knobLoadErr = errors.Newf("test error")
+	mu.Unlock()
+
 	producerJobID, ingestionJobID := c.StartStreamReplication(ctx)
 	jobutils.WaitForJobToRun(c.T, c.SrcSysSQL, jobspb.JobID(producerJobID))
 	jobutils.WaitForJobToRun(c.T, c.DestSysSQL, jobspb.JobID(ingestionJobID))
@@ -1150,24 +1155,13 @@ func TestTenantStreamingRetryLoadJob(t *testing.T) {
 	srcTime := c.SrcCluster.Server(0).Clock().Now()
 	c.WaitUntilReplicatedTime(srcTime, jobspb.JobID(ingestionJobID))
 
-	c.SrcSysSQL.Exec(t, fmt.Sprintf("PAUSE JOB %d", producerJobID))
-	jobutils.WaitForJobToPause(t, c.SrcSysSQL, jobspb.JobID(producerJobID))
-
-	// Write a bit more to be verified at the end.
-	c.SrcTenantSQL.Exec(t, "INSERT INTO d.t2 VALUES (3);")
-
-	// Inject an error to fail the resumer.
-	mu.Lock()
-	knobLoadErr = errors.Newf("test error")
-	mu.Unlock()
-
-	// Resume ingestion.
-	c.SrcSysSQL.Exec(t, fmt.Sprintf("RESUME JOB %d", producerJobID))
-	jobutils.WaitForJobToRun(t, c.SrcSysSQL, jobspb.JobID(producerJobID))
-
 	// Wait for the resumer to see the error and clear it, after this it should
 	// succeed resuming.
 	<-knobDoneCh
+	require.NoError(t, knobLoadErr)
+
+	// Write a bit more to be verified at the end.
+	c.SrcTenantSQL.Exec(t, "INSERT INTO d.t2 VALUES (3);")
 
 	// Verify the job succeeds now.
 	srcTime = c.SrcCluster.Server(0).Clock().Now()

